app/lib/dartdoc/backend.dart

```diff
@@ -168,44 +168,55 @@ class DartdocBackend {
     ]);
   }
 
+  /// Return the latest entries that should be used to serve the content.
+  ///
+  /// If the entry is missing for the current runtimeVersion, the method will
+  /// try to load the relevant entry for earlier runtimeVersions.
+  Future<List<DartdocEntry>> getEntriesForVersions(
+      String package, List<String> versions) async {
+    final entries = <String, DartdocEntry>{};
+
+    for (final v in versions) {
+      final cachedEntry = await cache.dartdocEntry(package, v).get();
+      if (cachedEntry != null) {
+        entries[v] = cachedEntry;
+      }
+    }
+
+    for (final rv in shared_versions.acceptedRuntimeVersions) {
+      final queryVersions =
+          versions.where((v) => !entries.containsKey(v)).toList();
+      if (queryVersions.isEmpty) break;
+
+      final reports = await scoreCardBackend.loadReportForAllVersions(
+        package,
+        queryVersions,
+        reportType: ReportType.dartdoc,
+        runtimeVersion: rv,
+      );
+      for (var i = 0; i < queryVersions.length; i++) {
+        final r = reports[i];
+        if (r != null) {
+          final version = queryVersions[i];
+          final entry = (r as DartdocReport).dartdocEntry;
+          entries[version] = entry;
+          await cache.dartdocEntry(package, version).set(entry);
```

Comment:
Technically, there is no need to await this here... I would suggest using the same pool as before.
```dart
scheduleMicrotask(() async {
  await pool.withResource(() async {
    await cache.dartdocEntry(package, version).set(entry);
  });
});
```